src/main/test/com/epam/preprod/sirenko/SafeContainerTest.java

```diff
@@ -0,0 +1,721 @@
+		container.add(food);
+		container.add(clothing);
+		containerToCheck.add(clothing);
+		containerToCheck.add(null);
+		
+		container.removeAll(containerToCheck);
+		
+		assertEquals(food, container.get(0));
+		assertEquals(food, container.get(1));
+		assertEquals(2, container.size());
+	}
+	
+	@Test
+	void removeAllShouldReturnFalseIfNotContainsElements() {
+		SafeContainer<Product> container = new SafeContainer<>();
+		SafeContainer<Product> containerToCheck = new SafeContainer<>();
+		Food food = new Food();
+		Food food1 = new Food();
+		Clothing clothing = new Clothing();
+		DryFood dryFood = new DryFood();
+		container.add(food);
+		container.add(food1);
+		containerToCheck.add(clothing);
+		containerToCheck.add(dryFood);
+		
+		container.removeAll(containerToCheck);
+		
+		assertEquals(food, container.get(0));
+		assertEquals(food1, container.get(1));
+	}
+	
+	@Test
+	void removeAllShouldReturnFalseIfNotContainsOnlyOneExistingElement() {
+		SafeContainer<Product> container = new SafeContainer<>();
+		SafeContainer<Product> containerToCheck = new SafeContainer<>();
+		Food food = new Food();
+		Clothing clothing = new Clothing();
+		container.add(clothing);
+		containerToCheck.add(food);
+		
+		container.removeAll(containerToCheck);
+		
+		assertEquals(clothing, container.get(0));
+	}
+	
+	@Test
+	void shouldReturnNextElement() {
+		SafeContainer<Product> container = new SafeContainer<>();
+		DryFood dryFood = new DryFood();
+		Clothing clothing = new Clothing();
+		container.add(dryFood);
+		container.add(clothing);
+		
+		Product product = container.iterator().next();
+		
+		assertEquals(product, dryFood);
+	}
+	
+	@Test
+	void shouldReturnTheOnlyOneExistingNextElement() {
+		SafeContainer<Product> container = new SafeContainer<>();
+		DryFood dryFood = new DryFood();
+		container.add(dryFood);
+		
+		Product product = container.iterator().next();
+		
+		assertEquals(product, dryFood);
+	}
+	
+	@Test
+	void shouldThrowNoSuchElementExceptionWhenNextElementNotExist() {
+		SafeContainer<Product> container = new SafeContainer<>();
+		
+		Iterator<Product> iterator = container.iterator();
+		
+		assertThrows(NoSuchElementException.class, () -> {
+			iterator.next();
+		});
+	}
+	
+	@Test
+	void shouldReturnFalseWhenNextElementNotExist() {
+		SafeContainer<Product> container = new SafeContainer<>();
+		
+		assertFalse(container.iterator().hasNext());
+	}
+	
+	@Test
+	void shouldReturnTrueIfHasNext() {
+		SafeContainer<Product> container = new SafeContainer<>();
+		Clothing clothing = new Clothing();
+		container.add(clothing);
+		
+		assertTrue(container.iterator().hasNext());
+	}
+	
+	@Test
+	void iteratorShouldUseUnchangedContainer() {
+		SafeContainer<Product> container = new SafeContainer<>();
+		Clothing clothing = new Clothing();
+		Food food = new Food();
+		DryFood dryFood = new DryFood();
+		container.add(clothing);
+		container.add(food);
+		container.add(dryFood);
+		Iterator<Product> iterator = container.iterator();
```

Comment:
(minor)Please add spaces to better visibility(given when then pattern for tests)